articles/understanding-tcp-ip-transport-layer-protocols/index.md

```diff
@@ -0,0 +1,164 @@
+# Understanding TCP/IP Transport Layer (Layer 4) protocols: TCP and UDP  
+
+![Hero image](/engineering-education/understanding-tcp-ip-transport-layer-protocols/hero-image.png)
+![Source](https://en.wikiversity.org/wiki/Internet_Protocol_Analysis/Transport_Layer#/media/File:Internet_Protocol_Analysis_-_Transport_Layer.png)
+
+### Introduction
+
+Each TCP/IP application typically choose to establish a temporary communication session to deliver data between two devices using either Transport Control Protocol (TCP) or User Datagram Protocol (UDP) based on the application requirements; this is done irrespective  regard to the destination host type, the type of media over which the data is to be sent, and the pathway taken by the information [Cisco](https://www.flashcardmachine.com/cisco-1-chapter4.html)
+TCP provides a wide range of services such as error recovery, windowing, and order segment delivery to applications, whereas UDP does not.
+In this article, we intend to look at these two transport layer protocols in detail.  
+
+### Prerequisites
+
+To better understand this article, foundational knowledge of TCP/IP  Layer 1 to 3 is required.  
+
+**Content**
+  -Transport Layer responsibilities  
+  -Multiplexing
+  -Reliability of the transport layer
+  -Transport Control Protocol (TCP)
+  -User Datagram Protocol (UDP)
+  -Multiple conversations
+  -Port number
+  -Socket and Socket pairs
+  -Conclusion
+
+### Transport Layer responsibility
+
+ The Transport layer  has a set of responsibilities to help applications function efficiently and adequately; these responsibilities are:
+-The tracking of individual conversation  
+-Ordered data transfer and data segmentation.
+-Identifying applications.
+
+**Tracking of Individual applications**
+Data flowing from one application to another is known as conversation.
+ Because a host can have multiple applications communicating with other applications either within the local network or with a remote network, the transport layer has a mechanism that makes it possible for each application on a host to communicate with another application on a different host either within the local network or in a remote network. According to [Cisco](https://community.cisco.com/t5/switching/which-layer-of-the-model-osi-and-tcp-ip-is-the-port-number/td-p/3356063 ), this mechanism assigns an identifier called a **port number** to each application such that each software process that needs to access the network is has a unique identifier.
```

Comment:
```suggestion
Because a host can have multiple applications communicating with other applications either within the local network or with a remote network, the transport layer has a mechanism that makes it possible for each application on a host to communicate with another application on a different host either within the local network or in a remote network. According to [Cisco](https://community.cisco.com/t5/switching/which-layer-of-the-model-osi-and-tcp-ip-is-the-port-number/td-p/3356063 ), this mechanism assigns an identifier called a **port number** to each application such that each software process that needs to access the network is has a unique identifier.
```